### PR TITLE
Use typed test collection names

### PR DESCRIPTION
**Pull request by amoerie**
(Hopefully) fixes some my OCD.

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [x] I have updated API documentation
- [x] I have included unit tests
- [x] ~~I have updated the change log~~
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Don't use magic strings when it comes to test collections
- Now it is easier to find usages of test collections
- Now it is harder to make typos

### CODE REVIEW

File: Tests/FO-DICOM.Tests/Bugs/GH306.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@
 namespace FellowOakDicom.Tests.Bugs
 {
 
-    [Collection("Network"), Trait("Category", "Network")]
+    [Collection(TestCollections.Network), Trait("Category", "Network")]
```
Comment (gofal) [inline Tests/FO-DICOM.Tests/Bugs/GH306.cs]: Also here `Trait(TestCollections.TraitCategory, TestCollections.TraitNetwork)`, to make the collection "Network" distinguishable from the trait "Network"?

File: Tests/FO-DICOM.Tests/Bugs/GH426.cs [hunk 2]
```diff
@@ -12,7 +12,7 @@
 namespace FellowOakDicom.Tests.Bugs
 {
 
-    [Collection("Network"), Trait("Category", "Network")]
+    [Collection(TestCollections.Network), Trait("Category", "Network")]
```
Comment (gofal) [inline Tests/FO-DICOM.Tests/Bugs/GH426.cs]: also Traits?

**Comment by gofal** [comment]
👍

File: Tests/FO-DICOM.Tests/Bugs/GH306.cs (hunk 1 above)
Comment (amoerie) [reply Tests/FO-DICOM.Tests/Bugs/GH306.cs]: I put the strings in TestTraits and TestCategories respectively, even though it feels a bit overkill when you realize that Trait("Network", "Category") is the only combination of parameters we use, ever. :-)

Comment (gofal) [reply Tests/FO-DICOM.Tests/Bugs/GH306.cs]: Oh, this is the only Trait-combination, that is used? Is it an idea to remove it, if there is no use of this Trais property?

**Review by gofal — APPROVED** [review]

**Pull request merged**